libbindgen/src/clang.rs

```diff
@@ -362,11 +362,12 @@ impl Cursor {
 
     /// Get the integer representation type used to hold this cursor's referent
     /// enum type.
-    pub fn enum_type(&self) -> Type {
+    pub fn enum_type(&self) -> Option<Type> {
         unsafe {
-            Type {
+            let t = Type {
                 x: clang_getEnumDeclIntegerType(self.x),
-            }
+            };
+            if t.kind() == CXType_Invalid { None } else { Some(t) }
```

Comment:
also, please use `if t.is_valid()` instead of checking `t.kind()` manually, just in case we want to eventually change the behavior of `is_valid()`.

Thanks!